### PR TITLE
Housekeeping: Rename variables to avoid collisions with packages

### PR DESCRIPTION
**Pull request by vdavid**
This is a refactor PR; it shouldn't change behavior.

A bunch of variables in the codebase collided with package names.
This can cause confusion and unwanted behavior that increases debugging time during development.

I've renamed the colliding variables to make them unique.
I've tried to improve code quality with this change by making the variable names at least as descriptive as they were while embracing the conciseness that seems idiomatic in Go. It was, at times, difficult to strike a balance between these, but at this point, I'm satisfied with the result.
In some cases, I opted for not renaming the variable names at all but rather adding an alias for the package with the conflicting name. I've committed those changes in a separate PR: https://github.com/sourcegraph/sourcegraph/pull/47180.

**For reviewers:** The recommended way to review this is to just scroll through it, either in the unified or the split view, whichever you're more used to. That's what I did about twice. It is long, but it's extremely simple: 100% renames. It will take less than 10 minutes if you agree with all my changes, a bit more otherwise. It needs some concentration, though—naming is hard. Thanks for loaning me your brain on this! 🧠 

## Test plan

Refactor commit with only renames. Tests should pass.

### CODE REVIEW

File: cmd/blobstore/shared/shared.go [hunk 1]
```diff
@@ -60,14 +60,14 @@ func Start(ctx context.Context, observationCtx *observation.Context, config *Con
 	// Ready immediately
 	ready()
 
-	service := &blobstore.Service{
+	bsService := &blobstore.Service{
```
Comment (eseliger) [inline cmd/blobstore/shared/shared.go:63]: cmon this service ain't _that_ bad 😆

**Review by eseliger — APPROVED** [review]

**Comment by unknwon** [comment]
Just dropping it as a data point 😁 

Do not fully agree with the reasoning, but also my opinion is not strong enough for an objection, but I do feel like we are doing X for the sake of X.

**Comment by vdavid** [comment]
@unknwon thank you, that's a useful data point! What would be your solution? Would you—like I did—continually ignore the warnings, _or_ would you turn off this type of warning in the IDE, deeming them unuseful?
On my end, I considered them useful but ignoring them was inconvenient, and that's why I opted for getting rid of them.

**Comment by unknwon** [comment]
> @unknwon thank you, that's a useful data point! What would be your solution? Would you—like I did—continually ignore the warnings, _or_ would you turn off this type of warning in the IDE, deeming them unuseful? On my end, I considered them useful but ignoring them was inconvenient, and that's why I opted for getting rid of them.

@vdavid - I use GoLand so it does always prompt/warn me about naming collisions, so reading code within the IDE would be visually informed that the name is reused, which is like five-9s fine for me personally, and I treat these warnings as FYI and just ignore them. 

Reading on not-yet-powerful-enough tools (like SG ourselves 😂) though, does have the possibility of creating confusion while reading the code, but I can't recall when was the last time I was confused by naming collisions but I'd definitely blame the tool first.

Maybe this is not always possible, but I think my personal principle is "use a name as long as it makes sense (convey the meaning in the best way within the context of the code block), and deal with naming collisions case-by-case (i.e. I have a variable `url` but also want to import `net/url`)".

**Pull request merged**